packages/display_renderers/src/robot_info_renderer_node.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import rospy
+import requests
+from typing import Dict
+
+from dt_robot_utils import get_robot_configuration
+from duckietown_msgs.msg import DisplayFragment as DisplayFragmentMsg
+
+from display_renderer import (
+    PAGE_ROBOT_INFO,
+    REGION_BODY,
+    DisplayROI,
+    TextFragmentRenderer,
+)
+
+from duckietown.dtros import DTROS, NodeType, TopicType
+
+
+class RobotInfoRendererNode(DTROS):
+    def __init__(self):
+        super(RobotInfoRendererNode, self).__init__(
+            node_name="robot_info_renderer_node", node_type=NodeType.VISUALIZATION
+        )
+        # get parameters
+        self._veh = rospy.get_param("~veh")
+        self._frequency = rospy.get_param("~frequency")
+        # create publisher
+        self._pub = rospy.Publisher(
+            "~fragments",
+            DisplayFragmentMsg,
+            queue_size=1,
+            dt_topic_type=TopicType.VISUALIZATION,
+            dt_help="Fragments to display on the display",
+        )
+        # create renderers
+        self._renderer = RobotInfoRenderer()
+        # create loop
+        self._timer = rospy.Timer(rospy.Duration.from_sec(1.0 / self._frequency), self._beat)
+
+    def _beat(self, _):
+        robot_info_str = self._fetch()
+        self._renderer.update(robot_info_str)
+        self._pub.publish(self._renderer.as_msg())
+
+    def _fetch(self) -> str:
+        health_api_url = f"http://{self._veh}.local/health/"
+        # noinspection PyBroadException
+        try:
+            health_data = requests.get(health_api_url).json()
+        except BaseException:
+            return
+
```

Comment:
Alternatively, use a busy wait instead of a Timer, then fall into `rospy.spin()` with the Publisher in latch mode.


```
health_data = None
while OK():
        time.sleep(5)
        try:
            health_data = requests.get(health_api_url).json()
        except BaseException:
            continue
       break

self._pub.publish(health_data)
rospy.spin()
```